src/components/NewPosition/RangeSelector/RangeSelector.tsx

```diff
@@ -249,10 +260,28 @@ export const RangeSelector: React.FC<IRangeSelector> = ({
       currentMidPrice !== midPrice &&
       !shouldReversePlot
     ) {
-      resetPlot()
-      setCurrentMidPrice(midPrice)
+      if (!shouldNotUpdatePriceRange) {
+        resetPlot()
+        setCurrentMidPrice(midPrice)
+      }
+    }
+  }, [triggerReset])
+
+  useEffect(() => {
+    if (
+      !ticksLoading &&
+      isMountedRef.current &&
+      poolKey !== '' &&
+      currentMidPrice !== midPrice &&
+      !shouldReversePlot
+    ) {
+      if (!shouldNotUpdatePriceRange) {
+        setTriggerReset(prev => !prev)
+      }
+
+      dispatch(positionsActions.setShouldNotUpdateRange(false))
```

Comment:
If possible, try to avoid dispatching actions within the component, and do it in the container instead